spec/active_force/active_query_spec.rb

```diff
@@ -463,4 +463,22 @@
    end
     
   end
+
+  describe '#first' do
+    before do
+      allow(client).to receive(:query).and_return(api_result)
+      api_result.each do |instance|
+        allow(active_query).to receive(:build).with(instance, {}).and_return(double(:sobject, id: instance['Id']))
+      end
+    end
+
+    it 'returns a single record when the api was already queried' do
+      active_query.to_a
```

Comment:
I'm not sure this spec proves anything? 

`expect(active_query.first.id).to eq("0000000000AAAAABBB")` is called in both, it's unclear what the `to_a` is supposed to prove here?

Reply:
over here, `to_a` is a way to get the instance variable `@records` to be populated on active_query. The `to_a` that runs via `super.to_a` is actually a different call, as that method is run on a separate cloned object that has the @records variable set

Reply:
Hmm, ok, I see yeah. Maybe a comment to explain this for those of us that don't know active_force that well or doesn't remember that `to_a` does what you said?